src/main/java/org/opensearch/geospatial/index/mapper/xypoint/XYPointParser.java

```diff
@@ -0,0 +1,175 @@
+     * @param ignoreZValue  boolean parameter which decides if third coordinate needs to be ignored or not
+     * @return {@link XYPoint} after setting the x and y coordinates parsed from the parse
+     * @throws IOException
+     * @throws OpenSearchParseException
+     */
+    public static XYPoint parseXYPoint(XContentParser parser, XYPoint point, final boolean ignoreZValue) throws IOException,
+        OpenSearchParseException {
+        double x = Double.NaN;
+        double y = Double.NaN;
+        NumberFormatException numberFormatException = null;
+
+        if (parser.currentToken() == XContentParser.Token.START_OBJECT) {
+            try (XContentSubParser subParser = new XContentSubParser(parser)) {
+                while (subParser.nextToken() != XContentParser.Token.END_OBJECT) {
+                    if (subParser.currentToken() != XContentParser.Token.FIELD_NAME) {
+                        throw new OpenSearchParseException("token [{}] not allowed", subParser.currentToken());
+                    }
+                    String field = subParser.currentName();
+                    if (!(X.equals(field) || Y.equals(field))) {
+                        throw new OpenSearchParseException("field must be either [{}] or [{}]", X, Y);
+                    }
+                    if (X.equals(field)) {
+                        subParser.nextToken();
+                        switch (subParser.currentToken()) {
+                            case VALUE_NUMBER:
+                            case VALUE_STRING:
+                                try {
+                                    x = subParser.doubleValue(true);
+                                } catch (NumberFormatException e) {
+                                    numberFormatException = e;
+                                }
+                                break;
+                            default:
+                                throw new OpenSearchParseException("x must be a number");
+                        }
+                    }
+                    if (Y.equals(field)) {
+                        subParser.nextToken();
+                        switch (subParser.currentToken()) {
+                            case VALUE_NUMBER:
+                            case VALUE_STRING:
+                                try {
+                                    y = subParser.doubleValue(true);
+                                } catch (NumberFormatException e) {
+                                    numberFormatException = e;
+                                }
+                                break;
+                            default:
+                                throw new OpenSearchParseException("y must be a number");
+                        }
+                    }
+                }
+            }
+            if (numberFormatException != null) {
+                throw new OpenSearchParseException("[{}] and [{}] must be valid double values", numberFormatException, X, Y);
+            }
```

Comment:
the message here is misleading. It will tell both x and y are not double even when 1 of them is double and another is not double. Please fix this.

Reply:
ack